### PR TITLE
tools(compiler): use closure compiler

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

File: build-system/tasks/compile.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+var closureCompiler = require('gulp-closure-compiler');
+var gulp = require('gulp');
+
+gulp.task('compile', function() {
```
Comment (cramforce) [inline build-system/tasks/compile.js]: Add some comments here from my PR.

File: 3p/polyfills.js [hunk 2]
```diff
@@ -20,4 +20,4 @@
 
 
 // This list should not get longer without a very good reason.
-import '../third_party/babel/custom-babel-helpers';
+//import '../third_party/babel/custom-babel-helpers';
```
Comment (cramforce) [inline 3p/polyfills.js]: We need to instead do this as a preprocess step.

**Comment by cramforce** [comment]
I think we should land this change.

Once I have changes for CC in we can also run it on travis to not regress.

One thing we need to fix is that the patch to polyfills.js needs to be applied programatically.

**Comment by erwinmombay** [comment]
@cramforce closing this in favor of your PR https://github.com/ampproject/amphtml/pull/1218

**Pull request closed without merging**